src/stores/user-store.js

```diff
@@ -34,11 +34,24 @@ class UserStore extends EventEmitter {
         this._users.push(user);
     }
 
+    _updateUser (user) {
+        const index = this._users.findIndex((oldUser) => user.id === oldUser.id);
+        if (index >= 0) {
```

Comment:
(minor)
The good practice is to add in error's message all information that could be helpful to detect the error, something like
```js
    throw new Error(`User with this ${user.id} not found.`)
```